### PR TITLE
fix: transition time status display issue

### LINKED ISSUE #255: Embedder error status display
embedder.status.conditions[0].lastTransitionTime and lastSuccessfulTime always display the current time.

### PR DESCRIPTION
**Pull request by Lanture1064**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes

Fixes #255

### CODE REVIEW

File: controllers/embedder_controller.go [hunk 1]
```diff
@@ -157,14 +157,26 @@ func (r *EmbedderReconciler) UpdateStatus(ctx context.Context, instance *arcadia
...
+			}
+		} else {
+			// Refresh successful time only
+			newCondition = arcadiav1alpha1.Condition{
+				Type:               arcadiav1alpha1.TypeReady,
+				Status:             corev1.ConditionTrue,
+				Reason:             arcadiav1alpha1.ReasonAvailable,
+				Message:            msg,
+				LastTransitionTime: instanceCopy.Status.GetCondition(arcadiav1alpha1.TypeReady).LastTransitionTime,
+				LastSuccessfulTime: metav1.Now(),
```
Comment (nkwangleiGIT) [inline controllers/embedder_controller.go:178]: seems most of the code are duplicated, we can use a single object and only use different LastSuccessfulTime if it's the only difference.

Comment (nkwangleiGIT) [inline controllers/embedder_controller.go:178]: seems most of the code are duplicated, we can use a single object and only use different LastSuccessfulTime if it's the only difference.

File: controllers/embedder_controller.go [hunk 2]
```diff
@@ -189,7 +189,7 @@ func (r *EmbedderReconciler) UpdateStatus(ctx context.Context, instance *arcadia
 	if err != nil {
 		// Set status to unavailable
 		newCondition = arcadiav1alpha1.Condition{
-			Type:               arcadiav1alpha1.TypeReady,
+			Type:               arcadiav1alpha1.TypeUnknown,
```
Comment (bjwswang) [inline controllers/embedder_controller.go:192]: @Lanture1064  Can you explain this change?

File: controllers/embedder_controller.go [hunk 3]
```diff
@@ -200,15 +200,18 @@ func (r *EmbedderReconciler) UpdateStatus(ctx context.Context, instance *arcadia
...
-		// Set status to available
 		newCondition = arcadiav1alpha1.Condition{
 			Type:               arcadiav1alpha1.TypeReady,
 			Status:             corev1.ConditionTrue,
 			Reason:             arcadiav1alpha1.ReasonAvailable,
 			Message:            msg,
-			LastTransitionTime: metav1.Now(),
 			LastSuccessfulTime: metav1.Now(),
 		}
+		if instanceCopy.Status.GetCondition(arcadiav1alpha1.TypeReady).Status != corev1.ConditionTrue {
```
Comment (bjwswang) [inline controllers/embedder_controller.go:210]: we have a IsReady() function https://github.com/kubeagi/arcadia/blob/main/api/v1alpha1/condition.go#L214

**Pull request closed without merging**